### PR TITLE
Removes mentions of styled system props from the CONTRIBUTING doc.

### PR DESCRIPTION
**Pull request by jfuchs**
More cleanup of stray styled system prop artifacts.

### Merge checklist

- [ ] Added/updated tests
- [x] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: contributor-docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -78,65 +78,41 @@ Additionally, every component should support [the `sx` prop](https://primer.styl
...
 
 ```jsx
-import {TYPOGRAPHY, COMMON} from './constants'
-import sx from './sx'
+import sx, {SxProp} from './sx'
 
-const Component = styled.div`
+const Component =
+  styled.div <
+  SxProp >
```
Comment (colebemis) [inline contributor-docs/CONTRIBUTING.md:85]: Is there a reason this is split across three lines?

Comment (jfuchs) [reply contributor-docs/CONTRIBUTING.md:85]: Oh strange! Prettier did that for me. I'll fix.

File: contributor-docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -78,65 +78,35 @@ Additionally, every component should support [the `sx` prop](https://primer.styl
...
-```
-
-Remember that the system prop function inside your style declaration needs to go _after_ any built-in styles you want to be overridable.
-
 ### Adding the `sx` prop
 
-Each component should provide access to a prop called `sx` that allows for setting theme-aware ad-hoc styles. See the [overriding styles](https://primer.style/components/overriding-styles) doc for more information on using the prop.
+Each component should accept a prop called `sx` that allows for setting theme-aware ad-hoc styles. See the [overriding styles](https://primer.style/components/overriding-styles) doc for more information on using the prop.
 
 Adding the `sx` prop is similar to adding system props; import the default export from the `sx` module, add it to your style definition, and add the appropriate prop types. **The `sx` prop should go at the _very end_ of your style definition.**
```
Comment (pksjce) [inline contributor-docs/CONTRIBUTING.md:101]: Maybe remove the reference to 'system props' here because that term no longer has meaning?

**Review by pksjce — APPROVED** [review]

File: contributor-docs/CONTRIBUTING.md (hunk 2 above)
Comment (jfuchs) [reply contributor-docs/CONTRIBUTING.md:101]: Good call!

File: contributor-docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -175,15 +145,16 @@ See [`src/__tests__/example.js`](src/__tests__/example.js) for examples of ways
 
 ### TypeScript support
 
-Several of the projects that consume Primer Components are written in TypeScript. Though Primer Components is not currently written in TS, we do export type definitions in order to make Primer Components compatible with other TS projects.
+Primer React is written in TypeScript, and we include type definitions in our built artifacts. To check types, run the `typecheck` script:
```
Comment (siddharthkp) [inline contributor-docs/CONTRIBUTING.md:148]: sorry!

```suggestion
Primer React is written in TypeScript, we include type definitions in our built artefacts. To check types, run the `typecheck` script:
```

Comment (siddharthkp) [reply contributor-docs/CONTRIBUTING.md:148]: ignore me please 😅

**Pull request merged**